### PR TITLE
More concise formatting for homogeneous lists

### LINKED ISSUE #34: Better display for list columns if all elements have the same type
e.g., nested data frames

From https://github.com/tidyverse/tibble/issues/117.

### PR DESCRIPTION
**Pull request by krlmlr**
Also shows lists in green.

Closes #34.

Current implementation:

screenshot from 2018-04-17 07-55-40

With `[[]]`:

screenshot from 2018-04-17 08-05-43

With `*`:

screenshot from 2018-04-17 08-06-55

With `<list()>`:

screenshot from 2018-04-17 08-07-47

### CODE REVIEW

File: R/type.R [hunk 1]
```diff
@@ -69,3 +69,8 @@ format_type_sum <- function(x) UseMethod("format_type_sum")
 format_type_sum.default <- function(x) {
   style_type(paste0("<", x, ">"))
 }
+
+#' @export
+format_type_sum.type_sum_list <- function(x) {
+  style_type(paste0("[", x, "]"))
```
Comment (krlmlr) [inline R/type.R]: The formatting for the capital is defined here.

**Comment by hadley** [comment]
I'd prefer to hold off on this for a bit - I have had some insights from working on BigQuery that will take a while to process

**Comment by krlmlr** [comment]
Also need to look at what to show for e.g. `lm` objects (which have a `dim()` of `NULL`).

**Comment by krlmlr** [comment]
@hadley: Would you like to revisit the output for condensed lists?

**Comment by krlmlr** [comment]
Superseded by #153.

**Pull request closed without merging**